substrate/frame/state-trie-migration/src/lib.rs

```diff
@@ -441,20 +445,64 @@ pub mod pallet {
 	#[pallet::pallet]
 	pub struct Pallet<T>(_);
 
+	/// Default implementations of [`DefaultConfig`], which can be used to implement [`Config`].
+	pub mod config_preludes {
+		use super::*;
+		use frame_support::derive_impl;
+
+		pub struct TestDefaultConfig;
+
+		#[derive_impl(frame_system::config_preludes::TestDefaultConfig as frame_system::DefaultConfig, no_aggregated_types)]
+		impl frame_system::DefaultConfig for TestDefaultConfig {}
+
+		#[frame_support::register_default_impl(TestDefaultConfig)]
+		impl DefaultConfig for TestDefaultConfig {
+			#[inject_runtime_type]
+			type RuntimeEvent = ();
+			#[inject_runtime_type]
+			type RuntimeHoldReason = ();
+		}
+	}
+
+	/// The reason for this pallet placing a hold on funds.
+	#[pallet::composite_enum]
+	pub enum HoldReason {
```

Comment:
re-reading this pallet, I don't see any instance where we store the deposit. The deposit is basically the amount that will be either free-ed or burnt on the spot. Given that, I suggest: 

- simplifying the code to hide the `hold -> slash_all_held` away
- use a single hold reason. I don't see any improvement being achieved by distinguishing between the hold reasons. For trace-ability, from the dispatchable, it is clear which signed migration is being used.

Reply:
ok, thank you, fixed here:
https://github.com/paritytech/polkadot-sdk/pull/1801/commits/9f2c695c86d4879fb3f7419fc26e3aa8e03d9911
https://github.com/paritytech/polkadot-sdk/pull/1801/commits/c07b5b58efa23e00c66e0bfb31c3dc5531716284